app/models/location.rb

```diff
@@ -40,36 +40,36 @@ class Location < ApplicationRecord
 
   geocoded_by :address
 
-  attr_accessor :country
-
   def address
-    [address_line1, zip_code, city, state,
-     ISO3166::Country.find_country_by_alpha2(country_code)].compact.join(', ')
+    [
+      address_line1, zip_code, city, state,
+      ISO3166::Country.find_country_by_alpha2(country_code)&.unofficial_names&.first
+    ].compact.join(', ')
   end
 
   def country
-    ISO3166::Country.find_country_by_alpha2(country_code).unofficial_names[0]
+    ISO3166::Country.find_country_by_alpha2(country_code)&.unofficial_names&.first
   end
 
   def convert_country_to_country_code
-    if country
-      country_alpha2 = ISO3166::Country.find_country_by_name(country)
-      country_code = country_alpha2.alpha2 if country_alpha2
-    end
+    return unless country
+
+    country_alpha2 = ISO3166::Country.find_country_by_name(country)
+    self.country_code = country_alpha2.alpha2 if country_alpha2
   end
```

Comment:
How about something like (untested):

```
def country_code=
  super(ISO3166::Country.find_country_by_name(country).alpha2)
end
```

Inspired by:
https://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Overwriting+default+accessors

Reply:
Actually I had to change now this country logic... Because I want to be able to do something like this `Location.create(country: "Switzerland)` and it will save a new Location with `country_code = 'CH'`. I'll add it in my next commit once I'm done with the new tests.

Reply:
I've tested this and it doesn't work the way I want it. But see my new `location.rb`, I've changed some things anyways.